plover/oslayer/osxkeyboardcontrol.py

```diff
@@ -285,18 +286,15 @@ def suppress_keyboard(self, suppressed_keys=()):
 # let's us iterate over full characters in the string.
 def characters(s):
     encoded = s.encode('utf-32-be')
-    characters = []
-    for i in xrange(len(encoded)/4):
+    for i in xrange(len(encoded) / 4):
         start = i * 4
         end = start + 4
         character = encoded[start:end].decode('utf-32-be')
         yield character
 
-native_utf16 = 'utf-16-le' if sys.byteorder == 'little' else 'utf-16-be'
-
 def set_string(event, s):
-    buf = s.encode(native_utf16)
-    CGEventKeyboardSetUnicodeString(event, len(buf) / 2, buf)
+    buf = Foundation.NSString.stringWithString_(s)
+    CGEventKeyboardSetUnicodeString(event, len(buf), buf)
```

Comment:
Does this automatically convert the NSString to a buffer of unichar for us? I'd have expected to need to do `string.getCharacters_range_()` with a buffer and an appropriately sized range to massage the data into the expected format, but I'm not terribly familiar with PyObjC.